### PR TITLE
Bluetooth: Host: Clear settings CF instead of CCC

### LINKED ISSUE #82558: bt_gatt_clear_cf clears ccc instead of cf
**Describe the bug**

when using Bluetooth in combination with CONFIG_BT_SETTINGS parameters like bonding keys etc. are stored in non volatile memory. When bonds are deleted also the non volatile settings should be cleaned up. However, after connecting and deleting bonds to several different devices the following error message appeared in our application log: 

`<err> bt_gatt: Unable to restore CF: no cfg left`

when reading the flash memory we found several entries to bt/cf with old mac addresses but the the corresponding entries of bt/ccc and bt/keys etc. were missing. We found that the function bt_gatt_clear_cf in gatt.c actually calls the bt_settings_delete_**ccc** rather than the bt_settings_delete_**cf**

**Expected behavior**

also obsolete cf setting entries are deleted. I guess the code should have been:
```
static int bt_gatt_clear_cf(uint8_t id, const bt_addr_le_t *addr)
{
	struct gatt_cf_cfg *cfg;

	cfg = find_cf_cfg_by_addr(id, addr);
	if (cfg) {
		clear_cf_cfg(cfg);
	}

	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
		return bt_settings_delete_cf(id, addr);
	}

	return 0;

}
```

**Impact**

after multiple pair/unpair actions the settings area will be polluted with bt/cf entries that do not correspond to any cfg. 

**Environment (please complete the following information):**
at least observed in the following releases but this code also seems to be in the latest develop
- Zephyr 3.7.0
- Zephyr 3.5.0

### PR DESCRIPTION
**Pull request by theob-pro**
See commits message for details.

Fix https://github.com/zephyrproject-rtos/zephyr/issues/82558

### CODE REVIEW

File: subsys/bluetooth/host/settings.c [hunk 1]
```diff
@@ -318,6 +318,10 @@ int bt_settings_store(const char *key, uint8_t id, const bt_addr_le_t *addr, con
 		}
 	}
 
+#if defined(CONFIG_BT_TESTING)
+	bt_testing_settings_store_hook(key_str, value, val_len);
+#endif
```
Comment (jhedberg) [inline subsys/bluetooth/host/settings.c:323]: Can this use `IF_ENABLED()`?

Comment (theob-pro) [reply subsys/bluetooth/host/settings.c:323]: We will get "implicit declaration of function" errors in applications that uses settings but without `CONFIG_BT_TESTING` :/

File: tests/bsim/bluetooth/host/gatt/settings_clear/test_scripts/run_settings_clear.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+SIM_LEN_US=$((2 * 1000 * 1000))
+
+test_exe="${BSIM_OUT_PATH}/bin/bs_${BOARD_TS}_${test_name}_prj_conf"
+
+cd ${BSIM_OUT_PATH}/bin
+
+Execute ./bs_2G4_phy_v1 -v=${verbosity_level} -s=${simulation_id} -D=2 -sim_length=${SIM_LEN_US} $@
+
+Execute "${test_exe}" -v=${verbosity_level} -s=${simulation_id} -d=0 -rs=420 -testid=server \
+		-flash="${simulation_id}_server.settings" -flash_rm -RealEncryption=1
```
Comment (aescolar) [inline tests/bsim/bluetooth/host/gatt/settings_clear/test_scripts/run_settings_clear.sh:24]: Maybe a stupid question, but why keep the flash in a file in the disk if you are removing it after terminating the execution? (if you don't set any of the -flash options, the flash is kept by default in the linux process heap and not mapped to the linux filesystem.

Comment (theob-pro) [reply tests/bsim/bluetooth/host/gatt/settings_clear/test_scripts/run_settings_clear.sh:24]: There is no stupid questions! :)
And indeed, there is no reasons to keep the file in the disk. It's my bad, I thought for some reasons that it was needed for using settings in BabbleSim 😅

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**